kitsune/questions/handlers.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import dataclass, field
+
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from kitsune.questions.models import Answer, Question
+from kitsune.sumo.handlers import AbstractChain, AccountHandler
+
+
+class SpamAAQHandler(AccountHandler):
+    """Handler for spam cleanup."""
+
+    def handle_account(self, data: dict) -> None:
+        """Handler to delete all spam questions and answers for a given user."""
+        questions = data["questions"]
+        answers = data["answers"]
+
+        spam_questions = questions.filter(is_spam=True).values_list("id", flat=True)
+        spam_answers = answers.filter(is_spam=True).values_list("id", flat=True)
+        Question.objects.filter(id__in=spam_questions).delete()
+        Answer.objects.filter(id__in=spam_answers).delete()
+
+
+class ArchivedProductAAQHandler(AccountHandler):
+    """Handler for archived product cleanup."""
+
+    def handle_account(self, data: dict) -> None:
+        """Handler to delete all forum posts for a given user for archived products."""
+        questions = data["questions"]
+        answers = data["answers"]
+
+        archived_questions = questions.filter(product__is_archived=True).values_list(
+            "id", flat=True
+        )
+        archived_answers = answers.filter(question__in=archived_questions).values_list(
+            "id", flat=True
+        )
+        Question.objects.filter(id__in=archived_questions).delete()
+        Answer.objects.filter(id__in=archived_answers).delete()
```

Comment:
I think these should be switched in terms of order because of the dependency of `archived_answers` on `archived_questions`?